tests/unit/plugins/modules/cloud/scaleway/test_scaleway_compute_private_network.py

```diff
@@ -0,0 +1,168 @@
+            }
+    return Response(None, info)
+
+
+def response_with_nics():
+    info = {"status": 200,
+            "body": ('{ "private_nics": [{'
+                     '"id": "c123b4cd-ef5g-678h-90i1-jk2345678l90",'
+                     '"private_network_id": "b589b4cd-ef5g-678h-90i1-jk2345678l90",'
+                     '"server_id": "c004b4cd-ef5g-678h-90i1-jk2345678l90",'
+                     '"mac_address": "02:00:00:00:12:23",'
+                     '"state": "available",'
+                     '"creation_date": "2022-03-30T06:25:28.155973+00:00",'
+                     '"modification_date": "2022-03-30T06:25:28.155973+00:00",'
+                     '"zone": "fr-par-1"'
+                     '}]}'
+                     )
+            }
+    return Response(None, info)
+
+
+def response_when_add_nics():
+    info = {"status": 200,
+            "body": ('{ "private_nics": {'
+                     '"id": "c123b4cd-ef5g-678h-90i1-jk2345678l90",'
+                     '"private_network_id": "b589b4cd-ef5g-678h-90i1-jk2345678l90",'
+                     '"server_id": "c004b4cd-ef5g-678h-90i1-jk2345678l90",'
+                     '"mac_address": "02:00:00:00:12:23",'
+                     '"state": "available",'
+                     '"creation_date": "2022-03-30T06:25:28.155973+00:00",'
+                     '"modification_date": "2022-03-30T06:25:28.155973+00:00",'
+                     '"zone": "fr-par-1"'
+                     '}}'
+                     )
+            }
+    return Response(None, info)
+
+
+def response_remove_nics():
+    info = {"status": 200,
+            "body": ''
+            }
+    return Response(None, info)
+
+
+def test_scaleway_private_network_without_arguments(capfd):
+    set_module_args({})
+    with pytest.raises(SystemExit) as results:
+        scaleway_compute_private_network.main()
+    out, err = capfd.readouterr()
+
+    assert not err
+    assert json.loads(out)['failed']
+
+
+def test_scaleway_add_nic(capfd):
+    os.environ['SCW_API_TOKEN'] = 'notrealtoken'
+    pnid = 'b589b4cd-ef5g-678h-90i1-jk2345678l90'
+    cid = 'c004b4cd-ef5g-678h-90i1-jk2345678l90'
+    url = 'servers/' + cid + '/private_nics'
+
+    set_module_args({"project": "a123b4cd-ef5g-678h-90i1-jk2345678l90",
+                     "state": "present",
+                     "region": "par1",
+                     "compute_id": cid,
+                     "private_network_id": pnid
+                     })
+
+    with patch.object(Scaleway, 'get') as mock_scw_get:
+        mock_scw_get.return_value = response_without_nics()
+        with patch.object(Scaleway, 'post') as mock_scw_post:
+            mock_scw_post.return_value = response_when_add_nics()
+            with pytest.raises(SystemExit) as results:
+                scaleway_compute_private_network.main()
+        mock_scw_post.assert_any_call(path=url, data={"private_network_id": pnid})
+    mock_scw_get.assert_any_call(url)
+
+    out, err = capfd.readouterr()
```

Comment:
Shouldn't you decode `out` and look whether it contains what you expect? (Same for the other tests below.)

Reply:
thanks, i add some check  in a8f82e8